components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -692,10 +703,25 @@ private LocalClaimResDTO getLocalClaimResDTO(LocalClaim localClaim) {
         }
         localClaimResDTO.setAttributeMapping(attributeMappingDTOs);
         localClaimResDTO.setProperties(mapToProperties(claimProperties));
-
+        localClaimResDTO.setAssociatedExternalClaims(fetchAssociatedExternalClaims(localClaim.getClaimURI()));
         return localClaimResDTO;
     }
 
+    private List<AssociatedExternalClaimDTO> fetchAssociatedExternalClaims(String localClaimURI) {
+
+        List<AssociatedExternalClaimDTO> associatedExternalClaimDTOS = new ArrayList<>();
+        try {
+            List<Claim> associatedExternalClaimList = getClaimMetadataManagementService().
+                    getMappedExternalClaimsForLocalClaim(localClaimURI, ContextLoader.getTenantDomainFromContext());
+            for (Claim claim: associatedExternalClaimList) {
+                associatedExternalClaimDTOS.add(getAssociatedExternalClaimResDTO(claim));
+            }
+            return associatedExternalClaimDTOS;
+        } catch (Exception e) {
```

Comment:
Why we have to catch a general exception? Can't we ` throw handleClaimManagementException(e, <error code>, <data>);` instead RuntimeException?